src/Avalonia.Controls/TabControl.cs

```diff
@@ -83,67 +100,82 @@ public Dock TabStripPlacement
             set { SetValue(TabStripPlacementProperty, value); }
         }
 
-        protected override IItemContainerGenerator CreateItemContainerGenerator()
+        /// <summary>
+        /// Gets or sets the data template used to display the content of the control.
+        /// </summary>
+        public IDataTemplate ContentTemplate
         {
-            // TabControl doesn't actually create items - instead its TabStrip and Carousel
-            // children create the items. However we want it to be a SelectingItemsControl
-            // so that it has the Items/SelectedItem etc properties. In this case, we can
-            // return a null ItemContainerGenerator to disable the creation of item containers.
-            return null;
+            get { return GetValue(ContentTemplateProperty); }
+            set { SetValue(ContentTemplateProperty, value); }
         }
 
-        protected override void OnTemplateApplied(TemplateAppliedEventArgs e)
+        /// <summary>
+        /// Gets or sets the currently selected content.
+        /// </summary>
+        /// <value>
+        /// The content of the selected.
```

Comment:
This doesn't make much sense: the content of the selected what? I'm guessing it should be "The content of the selected tab"?